tktermwidget/tkterm.py

```diff
@@ -93,70 +107,94 @@ def __init__(self, master: Misc, autohide: bool = True, *args, **kwargs):
 
         # Grid widgets
         self.text.grid(row=0, column=0, sticky="nsew")
-        self.xscroll.grid(row=1, column=0, sticky="ew")
+        if kwargs.get("wrap", "char") == "none":
+            self.xscroll.grid(row=1, column=0, sticky="ew")
         self.yscroll.grid(row=0, column=1, sticky="ns")
 
         # Create command prompt
         self.directory()
 
         # Set variables
-        self.index = 1
+        self.longsymbol = "\\" if not SYSTEM == "Windows" else "\\"
+        self.index, self.cursor = 1, self.text.index("insert")
         self.current_process: Popen | None = None
+        self.latest = self.cursor
+        self.longflag = False
+        self.longcmd = ""
 
-        # Bind events & tags
+        # Bind events
         self.text.bind("<Up>", self.up, add=True)
         self.text.bind("<Down>", self.down, add=True)
-        self.text.bind("<Left>", self.left, add=True)
         self.text.bind("<Return>", self.loop, add=True)
-        self.text.bind("<BackSpace>", self.left, add=True)
 
-        # TODO: Refactor the way we get output from subprocess
-        self.text.bind("<Control-KeyPress-c>", self.kill, add=True) # Isn't working
+        for bind_str in ("<Left>", "<BackSpace>"):
+            self.text.bind(bind_str, self.left, add=True)
+        for bind_str in ("<Return>", "<ButtonRelease-1>"):
+            self.text.bind(bind_str, self.updates, add=True)
+
+        self.text.bind("<Control-KeyPress-c>", self.kill, add=True)  # Isn't working
 
         # History recorder
-        self.history = open(HISTORY_PATH / "history.txt", "r+")
+        self.history = open(
+            HISTORY_PATH / "history.txt" if not filehistory else filehistory,
+            "r+",
+            encoding="utf-8",
+        )
+
         self.historys = [i.strip() for i in self.history.readlines() if i.strip()]
-        self.hi = len(self.historys) - 1
+        self.historyindex = len(self.historys) - 1
+
+    def updates(self, _: Event) -> None:
+        """Update cursor"""
+        self.cursor = self.text.index("insert")
+        if float(self.cursor) < float(self.latest):
+            self.text.bind("<KeyPress>", self.eat, True)
+            self.text.bind("<KeyPress-BackSpace>", self.eat, True)
+        elif float(self.cursor) >= float(self.latest):
+            self.text.unbind("<Key>")
+            self.text.unbind("<KeyPress-Backspace>")
+
+    def directory(self) -> None:
```

Comment:
My biggest pet peeve of all is one line methods, is there any way this could babe removed??

Reply:
Can you give a suggestion?

Reply:
But it is more easier than write ```self.text.insert("insert", os.getcwd())```

Reply:
When given the choice of a single repetitive line or a one line method I choose to just copy paste the long one.

Reply:
That's really a problem that I don't know what should I do now (lol)